tests/ha/single_pod_restart/test_data_pod_restart.py

```diff
@@ -1040,112 +1059,118 @@ def test_continuous_reads_during_pod_restart(self):
+        LOGGER.info("Step 2: Successfully shutdown data pod %s. Verified cluster and "
+                    "services states are as expected & remaining pods status is online.", pod_name)
 
-        remain_pod_list = list(filter(lambda x: x != pod_name, pod_list))
-        LOGGER.info("Step 4: Verify services status on remaining pods %s are in online state",
-                    remain_pod_list)
+        LOGGER.info("Step 3: Perform READ-Verify on data written in Step 1")
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix,
+                                                    skipwrite=True, skipcleanup=True,
+                                                    nclients=10, nsamples=10)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 3: Performed WRITEs/READs-Verify with variable sizes objects.")
 
-        resp = self.hlth_master_list[0].get_pod_svc_status(pod_list=remain_pod_list, fail=False)
-        LOGGER.debug("Response: %s", resp)
-        assert_utils.assert_true(resp[0], resp)
-        LOGGER.info("Step 4: Services of pod are in online state")
+        LOGGER.info("Step 4: Perform WRITEs/READs-Verify with variable object sizes.")
+        if CMN_CFG["dtm0_disabled"]:
+            self.test_prefix = 'test-34073-1'
 
-        LOGGER.info("Step 5: Perform WRITEs with variable object sizes. (0B - 512MB(VM)/5GB(HW))")
-        users = self.mgnt_ops.create_account_users(nusers=1)
-        self.test_prefix = 'test-34073'
-        self.s3_clean.update(users)
         resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
                                                     log_prefix=self.test_prefix, skipread=True,
                                                     skipcleanup=True, nclients=10, nsamples=10)
         assert_utils.assert_true(resp[0], resp[1])
-        LOGGER.info("Step 5: Performed WRITEs with variable sizes objects.")
+        LOGGER.info("Step 4: Performed WRITEs/READs-Verify with variable object sizes.")
 
-        LOGGER.info("Step 6: Perform READs and verify DI on the written data in background")
+        LOGGER.info("Step 5: Perform READs and verify DI on the data in background")
+        self.test_prefix = 'test-34073'
         args = {'s3userinfo': list(users.values())[0], 'log_prefix': self.test_prefix,
                 'nclients': 1, 'nsamples': 10, 'skipwrite': True, 'skipcleanup': True,
                 'output': output}
-
         thread = threading.Thread(target=self.ha_obj.event_s3_operation,
                                   args=(event,), kwargs=args)
         thread.daemon = True  # Daemonize thread
         thread.start()
-
-        LOGGER.info("Step 6: Successfully started READs and verified DI on the written data in "
+        if CMN_CFG["dtm0_disabled"]:
+            self.test_prefix = 'test-34073-1'
```

Comment:
This is little confusing here.

So if DTM is True, we will go with new buckets created after restart (self.test_prefix = 'test-34073-1') only thread1 is running.
So I this step 5 logic should go into if else right ?

Reply:
We will start reads on both healthy and dg mode written data. So thread with self.test_prefix = 'test-34073-1'  will go inside if, and thread with self.test_prefix = 'test-34073' will remain outside of the condition.